benchflow-dsl/src/main/scala/cloud/benchflow/dsl/definition/configuration/goal/explorationspace/explorationvalues/ExplorationValuesIntYamlProtocol.scala

```diff
@@ -62,10 +72,48 @@ object ExplorationValuesIntYamlProtocol extends DefaultYamlProtocol {
 
   }
 
-  def getList(range: List[Int]): List[Int] = {
+  private def getRangeList(range: List[Int]): Try[List[Int]] = {
+
+    if (range.size != 2) {
+
+      Failure(new BenchFlowDeserializationException("range must be specified as a List of 2 elements", new Throwable))
 
-    (range.head to range(1)).toList
+    } else {
+
+      Success((range.head to range(1)).toList)
+
+    }
 
   }
 
+  private def getStepList(range: List[Int], step: String): Try[List[Int]] = {
+
+    if (range.size != 2) {
+
+      Failure(new BenchFlowDeserializationException("range must be specified as a List of 2 elements", new Throwable))
+
+    } else {
+
+      val prefix = step.substring(0, 1)
+
+      val triedValue = Try(step.substring(1).toInt)
+
```

Comment:
You also need to validate the validity of the entire step definition, as discussed in a subsequent comment on the tests. Do not call and wait for a fail of the `RangeWithStep` function if you already know something it is not valid.